library/lua/script-manager.lua

```diff
@@ -88,28 +88,38 @@ local INSTALLED_MODS_PATH = 'data/installed_mods/'
 -- changes to the files)
 local MOD_PATH_ROOTS = {WORKSHOP_MODS_PATH, MODS_PATH, INSTALLED_MODS_PATH}
 
-local function get_mod_id_and_version(path)
+function get_mod_info(path)
     local idfile = path .. '/info.txt'
     local ok, lines = pcall(io.lines, idfile)
     if not ok then return end
-    local id, version
+    local id, numerical_version, name, steam_id, display_version
     for line in lines do
         if not id then
             _,_,id = line:find('^%[ID:([^%]]+)%]')
         end
-        if not version then
+        if not numerical_version then
             -- note this doesn't include the closing brace since some people put
             -- non-number characters in here, and DF only reads the leading digits
             -- as the numeric version
-            _,_,version = line:find('^%[NUMERIC_VERSION:(%d+)')
+            _,_,numerical_version = line:find('^%[NUMERIC_VERSION:(%d+)')
         end
+        if not display_version then
+            _,_,display_version = line:find('^%[DISPLAYED_VERSION:(%d+%.%d+)')
+        end
+        if not name then
+            _,_,name = line:find('^%[NAME:([^%]]+)%]')
+        end
+        if not steam_id then
+            _,_,steam_id = line:find('^%[STEAM_FILE_ID:(%d+)')
+        end
+
         -- note that we do *not* want to break out of this loop early since
         -- lines has to hit EOF to close the file
     end
-    return id, version
+    return id, numerical_version, name, steam_id, display_version
```

Comment:
this is too many return values. Moreover, as a public API, a function signature like this makes future updates difficult. it would be better to return the values in a map so we can extend this later without worrying about breaking existing callers.